Modified privacy policy page to the current version of EvalAI

Sanji515 opened the pull request:
- Added the scrollbar from the EvalAI project

- Toggle the show/hide for the `scroll to top` button when scroll is greater or lesser than 180 respectively

- Changed the highlight color of the content title and also changed css for the texts to look more better


**<p align="center">Screenshot 1</p>**


![Screenshot from 2019-06-14 21-55-32](https://user-images.githubusercontent.com/32524438/59523526-2fb17e00-8eef-11e9-88c1-d3b46257522c.png)


**<p align="center">Screenshot 2</p>**

![Screenshot from 2019-06-14 21-53-24](https://user-images.githubusercontent.com/32524438/59523445-ec570f80-8eee-11e9-8ee0-64d6ad441f53.png)

Sanji515 committed:
modified privacy policy page

Sanji515 committed:
minor change

Sanji515 committed:
fixed travis errors

Shekharrajak commented on src/app/components/privacy-policy/privacy-policy.component.scss:
```diff
@@ -36,21 +34,21 @@
       }
 
       li.scroll-selected {
-        color: $red-light;
```
you can directly change the red-light variable. Also I think bit darker color will be better to read the text.

Sanji515 replied:
@Shekharrajak done

Shekharrajak reviewed (commented):
I think this should be top of https://github.com/Cloud-CV/EvalAI-ngx/pull/111

Sanji515 committed:
suggested changes

Shekharrajak commented:
Please update the branch and resolve conflicts.

Sanji515 committed:
resolved conflicts

Sanji515 committed:
changed scrollIntoView to window.scrollTo

Sanji515 commented:
@Shekharrajak Please review

After merging, `scrollIntoView` was not working properly. The offset of navbar is not included so I've used `scrollTo` in order to give offset of the navbar to scroll at the right element

Sanji515 committed:
resolved lint errors

Sanji515 committed:
resolved conflicts after merging the contact us page PR

Sanji515 commented:
@Shekharrajak Please review

Shekharrajak reviewed (commented):
It looks good for now. But if you observe it properly then you will find that UI is not that much attractive and smooth.

Spaces can be removed (like around nav bar, footer, left sidebar, ...) You can try out the demo URL in different devices.

*

Sanji515 commented:
> Spaces can be removed (like around nav bar, footer, left sidebar, ...)

Current navbar's height is different from the navbar in the current version and also same for the footer, it will be changed and also I left the spaces around sidebar to make the main content to align in the center to look more better

please let me know what are your thoughts, I can change if you want

Sanji515 commented:
> You can try out the demo URL in different devices

I have checked in the browser for different devices, a/c to me it's working perfectly, please check if it's looks good to you as well

lunayach commented:
@Sanji515 could you add some unit tests so that we merge this?

Sanji515 committed:
Add unit tests for some cases

- Added tests for the length of scroll element after when component initialized.
- Added tests for scroll element should have the first title name
- Added tests for the scroll function called when li.scroll-element is clicked
- Added tests for the highlight text functions get called when scroll function is called

Sanji515 committed:
Merge branch 'master' into privacy-policy-page

lunayach commented on src/app/components/privacy-policy/privacy-policy.component.spec.ts:
```diff
@@ -68,4 +79,24 @@ describe('PrivacyPolicyComponent', () => {
+    scrollSelected.nativeElement.click();
+
+    fixture.whenStable().then(() => {
+      expect(component.highlightNav).toHaveBeenCalled();
```
Repeated.

lunayach commented on src/app/components/privacy-policy/privacy-policy.component.spec.ts:
```diff
@@ -68,4 +79,24 @@ describe('PrivacyPolicyComponent', () => {
+
+    fixture.whenStable().then(() => {
+      expect(component.highlightNav).toHaveBeenCalled();
+      expect(component.highlightSectionTitle).toHaveBeenCalled();
```
This could have been written along with the above two "expect"s.

Sanji515 replied:
@lunayach  done

Sanji515 committed:
moved the scroll and hightlight function tests to the same block

Sanji515 committed:
Merge branch 'privacy-policy-page' of https://github.com/Sanji515/EvalAI-ngx into privacy-policy-page

lunayach reviewed (approved):
Looks good to me.
